src/test/scala/com/softwaremill/realworld/users/UsersEndpointsSpec.scala

```diff
@@ -80,14 +94,99 @@ object UsersEndpointsSpec extends ZIOSpecDefault:
           )
         )
       }
-    ) @@ TestAspect.before(withAuthData())
+    ) @@ TestAspect.before(withEmptyDb())
+      @@ TestAspect.after(clearDb),
+    suite("User register")(
+      test("return already in use error") {
+        assertZIO(
+          ZIO
+            .service[UsersEndpoints]
+            .map(_.userRegister)
+            .flatMap { endpoint =>
+              val backendStub =
+                zioTapirStubInterpreter
+                  .whenServerEndpoint(endpoint)
+                  .thenRunLogic()
+                  .backend()
+              basicRequest
+                .post(uri"http://test.com/api/users")
+                .body(UserRegister(UserRegisterData(email = "admin@example.com", username = "user", password = "password")))
+                .response(asJson[User])
+                .send(backendStub)
+                .map(_.body)
+            }
+        )(isLeft(equalTo(HttpError("{\"error\":\"E-mail already in use!\"}", sttp.model.StatusCode(409)))))
+      }
+      /*test("return registered user") { // TODO: usingRecursiveComparison + skip token field
```

Comment:
todo?

Reply:
yep, I'll cover this later. I'm trying to do something like "usingRecursiveComparison" (from AssertJ) and ignore token field in the comparison but without success so far :)

for now token is compared as well so this test would fail (this is why I commented it out)

Reply:
maybe you need sth like https://diffx-scala.readthedocs.io/en/latest/?

Reply:
I'll check it, thanks.